db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TupleDescriptor.java

```diff
@@ -71,6 +73,12 @@ public TupleDescriptor(DataDictionary dataDictionary) {
 	}
 
 	public DataDictionary getDataDictionary() {
+		if (dataDictionary == null) {
+			LanguageConnectionContext lcc =
+				(LanguageConnectionContext) ContextService.getContext(LanguageConnectionContext.CONTEXT_ID);
+			if (lcc != null)
+                dataDictionary = lcc.getDataDictionary();
+		}
```

Comment:
I think we should enforce dataDictionary is always set for any `TupleDescriptor` instead.

Reply:
OK, I will revert this change and set the dataDictionary in TriggerDescriptor.readExternal and SPSDescriptor.readExternal.  Since I'm serializing more items with this Jira, upon deserialization the dataDictionary link needs to be re-established.  It's possible that more descriptors might need this change with this approach.  I suppose testing will point that out.